libnest/src/repository/cache.rs

```diff
@@ -122,6 +158,27 @@ impl CategoryCache {
     /// # Filesystem
     ///
     /// This operation assumes the current user has the rights to read the local cache.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # extern crate libnest;
+    /// use libnest::config::Config;
+    /// use libnest::repository::Repository;
+    ///
+    /// let config = Config::new();
+    ///
+    /// let repository = Repository::new("stable");
```

Comment:
You should use `and_then()` here. It will make the code easier to read.

```rust
let manifest = repository.cache(&config).category("shell")
    .and_then(|category| category.manifest("bash"));

if let Ok(manifest) = manifest {
    println!("There is a package named bash within the shell category");
} else {
    println!("There is no such package");
}